src/Modal.js

```diff
@@ -315,4 +349,37 @@ const Modal = React.createClass({
   }
 });
 
-export default Modal;
+const ModalOverlay = React.createClass({
+  propTypes: {
+    ...Portal.propTypes
+  },
+
+  defaultProps: {
+    show: null
+  },
+
+  render() {
+    let { show, ...props } = this.props;
+
+    let modal = (
+      <Modal {...props}>{this.props.children}</Modal>
+    );
+    // I can't think of another way to not break back compat while defaulting container
+    if ( show != null ){
+      return (
+        <Portal show={show} container={props.container} >
+          { modal }
+        </Portal>
+      );
+    } else {
+      return modal;
+    }
+  }
+});
+
+ModalOverlay.Body = Body;
```

Comment:
both are fine, the factories will get the ModalBody, ModalTitle, etc components from lib. I find this way is just more pleasant and saves 5 additional imports for components that you will always use together. There is not prescription to do it this way its merely a convenience